### PR TITLE
Adding and removing volumes from JBOD storage

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

This PR adds the possibility to add or remove volumes in Jbod storage. The implementation is fairly simple - we just ignore added or removed volumes in the StorageDiff. However, that has a one particular caveat - the user can either add or remove volumes. But not both at the same time. I think this is reasonable trade-off for the simplicity of the code and if properly documented it should work fine for users.

An alternative would require to do the diffing on a lower level to identify replaced volumes (i.e. add + remove in the same change). That would make it much harder to maintain in the future if we add more options such as IOPS, Ephemeral volume sizes etc. So I would probably prefer to keep the simpler way.

This PR also adds check that the Jbod storage has at least one volume defined.

Docs will be delivered in separate PR.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests tags=acceptance,regression

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=LogSettingST,KafkaST,SecurityST,RollingUpdateST tags=acceptance,regression

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 1]
```diff
@@ -454,6 +454,10 @@ protected void setStorage(Storage storage) {
         } else if (storage instanceof JbodStorage)  {
             JbodStorage jbodStorage = (JbodStorage) storage;
 
+            if (jbodStorage.getVolumes().size() == 0)   {
+                throw new InvalidResourceException("JbodStorage need to contain at least one volume!");
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: typo "needs"

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/StorageDiff.java [hunk 2]
```diff
@@ -35,6 +36,7 @@ public StorageDiff(Storage current, Storage desired) {
 
         boolean changesType = false;
         boolean changesSize = false;
+        boolean volumeAddedRemoved = false;
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/StorageDiff.java]: is this used somewhere?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java (hunk 1 above)
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: Fixed.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/StorageDiff.java (hunk 2 above)
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/StorageDiff.java]: Removed.

**Comment by scholzj** [comment]
@tombentley So, are you try it say that you want the code to detect it and throw a special error messages for that? I can have a look at it. But if we go through the pain of detecting that, we will simply enable it.

**Comment by scholzj** [comment]
@tombentley Ok, you pushed me ... and I think it should be working now for adding and removing volumes at the same time. Would you mind having a look? (changes in StorageDiff) ...

... and thanks for pushing me, sometimes I need it to be less lazy ;-).

**Pull request merged**